File: modules/@angular/core/test/spies.ts
import {ElementRef} from '@angular/core';
import {ChangeDetectorRef} from '@angular/core/src/change_detection/change_detection';
import {SpyObject} from '@angular/core/testing/testing_internal';
import {DomAdapter} from '@angular/platform-browser/src/dom/dom_adapter';

export class SpyChangeDetectorRef extends SpyObject {
  constructor() {
    super(ChangeDetectorRef);
    this.spy('detectChanges');
    this.spy('checkNoChanges');
  }
}

export class SpyIterableDifferFactory extends SpyObject {}

export class SpyElementRef extends SpyObject {
  constructor() { super(ElementRef); }
}

export class SpyDomAdapter extends SpyObject {
  constructor() { super(DomAdapter); }
}

File: modules/@angular/core/test/spies.dart
library core.spies;

import 'package:angular2/core.dart';
import 'package:angular2/src/core/change_detection/change_detection.dart';
import 'package:angular2/src/platform/dom/dom_adapter.dart';
import 'package:@angular/core/testing/testing_internal.dart';

@proxy
class SpyChangeDetectorRef extends SpyObject implements ChangeDetectorRef {}

@proxy
class SpyIterableDifferFactory extends SpyObject
    implements IterableDifferFactory {}

@proxy
class SpyElementRef extends SpyObject implements ElementRef {}

@proxy
class SpyDomAdapter extends SpyObject implements DomAdapter {}
